Libraries/src/Amazon.Lambda.Annotations/APIGateway/HttpResults.cs

```diff
@@ -0,0 +1,437 @@
+// by Visual Studio. Several APIs are used in this file that are not available in .NET Standard 2.0 but the source
+// generator doesn't use the methods that use those .NET APIs. Several areas in this file put stub implementations
+// for .NET Standard 2.0 to allow the types to be available in the source generator but the implementations
+// are not actually called in the source generator.
+
+
+namespace Amazon.Lambda.Annotations.APIGateway
+{
+    /// <summary>
+    /// The options used by the IHttpResults to serialize into the required format for the event source of the Lambda funtion.
+    /// </summary>
+    public class HttpResultSerializationOptions
+    {
+        /// <summary>
+        /// The API Gateway protocol format used as the event source.
+        /// </summary>
+        public enum ProtocolFormat { 
+            /// <summary>
+            /// Used when a function is defined with the RestApiAttribute.
+            /// </summary>
+            RestApi, 
+            /// <summary>
+            /// Used when a function is defined with the HttpApiAttribute.
+            /// </summary>
+            HttpApi 
+        }
+
+        /// <summary>
+        /// The API Gateway protocol version.
+        /// </summary>
+        public enum ProtocolVersion {
+            /// <summary>
+            /// V1 format for API Gateway Proxy responses. Used for functions defined with RestApiAttribute or HttpApiAttribute with explicit setting to V1.
+            /// </summary>
+            V1, 
+            /// <summary>
+            /// V2 format for API Gateway Proxy responses. Used for functions defined with HttpApiAttribute with an implicit version of an explicit setting to V2.
+            /// </summary>
+            V2 
+        }
+
+        /// <summary>
+        /// The API Gateway protocol used as the event source. 
+        ///     RestApi -> RestApiAttrbute
+        ///     HttpApi -> HttpApiAttribute
+        /// </summary>
+        public ProtocolFormat Format { get; set; }
+
+        /// <summary>
+        /// The API Gateway protocol version used as the event source. 
+        ///     V1 -> RestApi or HttpApi specifically set as V1
+        ///     V2 -> HttpApi either implicit or explicit set to V2
+        /// </summary>
+        public ProtocolVersion Version { get; set; }
+    }
+
+    /// <summary>
+    /// If this inteface is returned for an API Gateway Lambda function it will serialize itself to the correct JSON format for the 
+    /// configured event source's protocol format and version.
+    /// 
+    /// Users should use the implementation class HttpResults to construct an instance of IHttpResults with the configured, status code, response body and headers.
+    /// </summary>
+    /// <example>
+    /// return HttpResults.Ok("All Good")
+    ///                   .AddHeader("Custom-Header", "FooBar");
+    /// 
+    /// </example>
+    public interface IHttpResults
```

Comment:
Not a big deal/okay if others disagree, but I don't love this as plural since when I look at the Lambda method signature I read it as _returning multiple http results_. 

Plus I think singular would more closely align with ASP.NET's `IResult`.

But I do like `HttpResults` as plural, because I think of `HttpResults.Ok` as I'm choosing `Ok` out of all of the possible results.

Reply:
I agree the interface name doesn't make sense to be plural. I renamed it to be `IHttpResult`.